yii2/modules/schooltransport/models/Statistic.php

```diff
@@ -0,0 +1,344 @@
+<?php
+namespace app\modules\schooltransport\models;
+
+use app\modules\schooltransport\Module;
+use DateTime;
+use Yii;
+use yii\base\Model;
+
+class Statistic extends Model
+{
+    const GROUPBY_YEAR = 'byYear';
+    const GROUPBY_COUNTRY = 'byCountry';
+    const GROUPBY_PERFECTURE = 'byPerfecture';
+    const GROUPBY_EDULEVEL = 'byEdulevel';
+    const GROUPBY_PROGRAM = 'byProgram';
+    
+    const CHARTTYPE_BAR = 'bar';
+    const CHARTTYPE_HORIZONTALBAR = 'horizontalBar';
+    const CHARTTYPE_PIE = 'pie';
+    const CHARTTYPE_DOUGHNUT = 'doughnut';
+    const CHARTTYPE_POLARAREA = 'polarArea';
+    
+    public $statistic_schoolyear;
+    public $statistic_country;
+    public $statistic_prefecture;
+    public $statistic_educationlevel;
+    public $statistic_program;
+    public $statistic_groupby;
+    public $statistic_charttype;
+    
+    /**
+     * @inheritdoc
+     */
+    public function rules()
+    {
+        return [[['statistic_schoolyear', 'statistic_country', 'statistic_prefecture',
+                  'statistic_educationlevel', 'statistic_program'], 'required'],
+                [['statistic_schoolyear'], 'each', 'rule' => ['integer']],
+                [['statistic_country'], 'string', 'max' => 100],
+                [['statistic_prefecture'], 'string', 'max' => 100],
+                [['statistic_educationlevel'], 'string', 'max' => 50],
+                [['statistic_program'], 'string', 'max' => 300],
+                [['statistic_groupby'], 'string', 'max' => 50],
+                [['statistic_charttype'], 'string', 'max' => 50]
+        ];
+    }
+
+/**
+ * @inheritdoc
+ */
+    
+    public function attributeLabels()
+    {
+        return ['statistic_schoolyear' => Module::t('modules/schooltransport/app', 'Σχολικό έτος'),
+                'statistic_country' => Module::t('modules/schooltransport/app', 'Χώρα Προορισμού'),
+                'statistic_prefecture' => Module::t('modules/schooltransport/app', 'Νομός Εκπαιδευτικής Μονάδας'),
+                'statistic_educationlevel' => Module::t('modules/schooltransport/app', 'Βαθμίδα Εκπαίδευσης'),
+                'statistic_program' => Module::t('modules/schooltransport/app', 'Πρόγραμμα'),
+                'statistic_groupby' => Module::t('modules/schooltransport/app', 'Ομοδοποίηση'),
+               ];
+    }
+    
+    /**
+     * Returns the description of the statistic as it has benn defined through the instance variables.
+     * 
+     * @return string
+     */
+    public function getStatisticLiteral(){
+        $years = $this->statistic_schoolyear;
+        $years_count = count($years);
+        $literal = 'Πλήθος σχολικών μετακινήσεων';
```

Comment:
Μείξη Module::t και in-place translations. Πρόταση για χρήση named placeholders και plural form.